From 316be6b29e59bcc5c66de5418d733fb589cdd116 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Tue, 22 Sep 2020 12:24:08 -0700
Subject: [PATCH 1632/1851] media: xilinx: vtc: Fix warning for
 OVERFLOW_BEFORE_WIDEN

commit 011862bae06a8178cf8cd0120473b7f4be9fb655 from
https://github.com/Xilinx/linux-xlnx.git

This patch fixes the warning for OVERFLOW_BEFORE_WIDEN.

The actual warning is -
For xvtc_generator_start():
Event overflow_before_widen: 	Potentially overflowing expression
"config->fps * config->hsize" with type "unsigned int" (32 bits,
unsigned) is evaluated using 32-bit arithmetic, and then used in a
context that expects an expression of type "unsigned long" (64 bits,
unsigned).
Event remediation: 	To avoid overflow, cast either "config->fps" or
"config->hsize" to type "unsigned long".

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-vtc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/platform/xilinx/xilinx-vtc.c b/drivers/media/platform/xilinx/xilinx-vtc.c
index d69f48dfdac4..f71db1c344b7 100644
--- a/drivers/media/platform/xilinx/xilinx-vtc.c
+++ b/drivers/media/platform/xilinx/xilinx-vtc.c
@@ -185,7 +185,7 @@ int xvtc_generator_start(struct xvtc_device *xvtc,
 	if (!xvtc->has_generator)
 		return -ENXIO;
 
-	s_rate = config->fps * config->hsize * config->vsize;
+	s_rate = (unsigned long)config->fps * config->hsize * config->vsize;
 	ret = clk_set_rate(xvtc->xvip.clk, s_rate);
 	if (ret < 0)
 		return ret;
-- 
2.31.1

